### PR TITLE
Prevent applications without templates or aggregating methods to brea…

### PR DESCRIPTION
**Pull request by josejulio**
…k the server

 - It will silently ignore applications without aggregating method or email template
 - Logs errors in the template and aborts the email

### CODE REVIEW

**Comment by josejulio** [comment]
@pilhuhn I updated this PR to remove the hard coded values of "insights" and "policies". This also prevents breaking if an app that does not have any email template tries to push a notification. (the email is silently ignored).
//cc: @theute

File: src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/EmailPayloadAggregator.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package com.redhat.cloud.notifications.processors.email.aggregators;
+
+import com.redhat.cloud.notifications.models.EmailAggregation;
+import io.vertx.core.json.JsonObject;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+public abstract class EmailPayloadAggregator {
```
Comment (pilhuhn) [inline src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/EmailPayloadAggregator.java:9]: Should we put 'Abstract' in the class name to make it more obvious when glancing at files?

File: src/main/java/com/redhat/cloud/notifications/templates/EmailTemplate.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.redhat.cloud.notifications.templates;
+
+import com.redhat.cloud.notifications.models.EmailSubscription.EmailSubscriptionType;
+import io.quarkus.qute.TemplateInstance;
+
+public abstract class EmailTemplate {
```
Comment (pilhuhn) [inline src/main/java/com/redhat/cloud/notifications/templates/EmailTemplate.java:6]: Likewise here 'AbstractEmailTemplate'

Comment (josejulio) [reply src/main/java/com/redhat/cloud/notifications/templates/EmailTemplate.java]: Just updated with the changes.

**Review by pilhuhn — APPROVED** [review]

**Pull request merged**